crates/blockchain/payload.rs

```diff
@@ -0,0 +1,129 @@
+use std::cmp::min;
+
+use ethereum_rust_core::{
+    types::{
+        calculate_base_fee_per_gas, compute_receipts_root, compute_transactions_root,
+        compute_withdrawals_root, Block, BlockBody, BlockHash, BlockHeader, Withdrawal,
+        DEFAULT_OMMERS_HASH,
+    },
+    Address, Bloom, Bytes, H256, U256,
+};
+use ethereum_rust_rlp::encode::RLPEncode;
+use ethereum_rust_storage::{error::StoreError, Store};
+use sha3::{Digest, Keccak256};
+
+use crate::constants::{GAS_LIMIT_BOUND_DIVISOR, MIN_GAS_LIMIT, TARGET_BLOB_GAS_PER_BLOCK};
+
+pub struct BuildPayloadArgs {
+    pub parent: BlockHash,
+    pub timestamp: u64,
+    pub fee_recipient: Address,
+    pub random: H256,
+    pub withdrawals: Vec<Withdrawal>,
+    pub beacon_root: Option<H256>,
+    pub version: u8,
+}
+
+impl BuildPayloadArgs {
+    /// Computes an 8-byte identifier by hashing the components of the payload arguments.
```

Comment:
this is what the spec says? not against it, just curious

Reply:
I think I took it from geth. The spec says `8 Bytes - identifier of the payload build process`